### ISSUE TITLE
Windows Setup Error

### ISSUE THREAD

**Issue by markkhoo** [sfbrigade/intentional-walk#258]
npm WARN ERESOLVE overriding peer dependency
npm WARN While resolving: react-localization@1.0.15  
npm WARN Found: react@17.0.1
npm WARN node_modules/react
npm WARN   react@"17.0.1" from the root project      
npm WARN   13 more (@react-native-community/art, ...)
npm WARN
npm WARN Could not resolve dependency:
npm WARN peer react@"^16.0.0 || ^15.6.0" from react-localization@1.0.15
npm WARN node_modules/react-localization
npm WARN   react-localization@"^1.0.14" from react-native-localization@2.1.6
npm WARN   node_modules/react-native-localization
npm WARN
npm WARN Conflicting peer dependency: react@16.14.0
npm WARN node_modules/react
npm WARN   peer react@"^16.0.0 || ^15.6.0" from react-localization@1.0.15
npm WARN   node_modules/react-localization
npm WARN     react-localization@"^1.0.14" from react-native-localization@2.1.6
npm WARN     node_modules/react-native-localization
npm ERR! code ERESOLVE
npm ERR! ERESOLVE could not resolve
npm ERR!
npm ERR! While resolving: react-moment@0.9.7
npm ERR! Found: react@17.0.1
npm ERR! node_modules/react
npm ERR!   react@"17.0.1" from the root project
npm ERR!   peer react@"*" from @react-native-community/art@1.1.2
npm ERR!   node_modules/@react-native-community/art
npm ERR!     @react-native-community/art@"^1.0.3" from react-native-progress@4.0.3
npm ERR!     node_modules/react-native-progress
npm ERR!       react-native-progress@"^4.0.3" from the root project
npm ERR!   12 more (@react-native-community/masked-view, ...)
npm ERR!
npm ERR! Could not resolve dependency:
npm ERR! peer react@"^15.6.0 || ^16.0.0" from react-moment@0.9.7
npm ERR! node_modules/react-moment
npm ERR!   react-moment@"^0.9.7" from the root project
npm ERR!
npm ERR! Conflicting peer dependency: react@16.14.0
npm ERR! node_modules/react
npm ERR!   peer react@"^15.6.0 || ^16.0.0" from react-moment@0.9.7
npm ERR!   node_modules/react-moment
npm ERR!     react-moment@"^0.9.7" from the root project
npm ERR!
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR!
npm ERR!
npm ERR! For a full report see:
npm ERR! C:\Users\Kimar\AppData\Local\npm-cache\_logs\2023-01-19T03_51_20_123Z-eresolve-report.txt

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\Kimar\AppData\Local\npm-cache\_logs\2023-01-19T03_51_20_123Z-debug-0.log

**Comment by markkhoo** [comment]
npm 9.2.0
node 18.12.1

**Comment by markkhoo** [comment]
**Metro Error:**

Failed to construct transformer:  Error: error:0308010C:digital envelope routines::unsupported
    at new Hash (node:internal/crypto/hash:71:19)
    at Object.createHash (node:crypto:133:10)
    at stableHash (C:\Users\Kimar\Projects\intentional-walk\node_modules\metro-cache\src\stableHash.js:19:8)
    at Object.getCacheKey (C:\Users\Kimar\Projects\intentional-walk\node_modules\metro-transform-worker\src\index.js:647:7)
    at getTransformCacheKey (C:\Users\Kimar\Projects\intentional-walk\node_modules\metro\src\DeltaBundler\getTransformCacheKey.js:32:19)
    at new Transformer (C:\Users\Kimar\Projects\intentional-walk\node_modules\metro\src\DeltaBundler\Transformer.js:180:28)
    at C:\Users\Kimar\Projects\intentional-walk\node_modules\metro\src\Bundler.js:58:29
    at runNextTicks (node:internal/process/task_queues:60:5)
    at process.processTimers (node:internal/timers:504:9) {
  opensslErrorStack: [ 'error:03000086:digital envelope routines::initialization error' ],
  library: 'digital envelope routines',
  reason: 'unsupported',
  code: 'ERR_OSSL_EVP_UNSUPPORTED'
}

**Comment by francisli** [comment]
@markkhoo I see:

```
'adb' is not recognized as an internal or external command,
```

This means Android command line tools are not in your path...?

**Comment by markkhoo** [comment]
Yea I am digging through it right now. It looks like the location of the Java Installtion has changed since this [issue](https://github.com/sfbrigade/intentional-walk/issues/204). I've changed the path variable. Will update soon.

**Comment by markkhoo** [comment]
The path variable does need to be updated in this [branch](https://github.com/sfbrigade/intentional-walk/tree/windowsReadMe).
`C:\Program Files\Android\Android Studio\jre` to `C:\Program Files\Android\Android Studio\jbr` if Android Studio is up to date at this time and installed in the default location. This seems to solve that issue.

However I get a new error:

**Comment by markkhoo** [comment]
FAILURE: Build failed with an exception.

* What went wrong:
A problem occurred configuring project ':react-native-reanimated'.
> java.lang.NullPointerException (no error message)

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 6s

error Failed to install the app. Make sure you have the Android development environment set up: https://reactnative.dev/docs/environment-setup.
Error: Command failed: gradlew.bat app:installDebug -PreactNativeDevServerPort=8081

FAILURE: Build failed with an exception.

* What went wrong:
A problem occurred configuring project ':react-native-reanimated'.
> java.lang.NullPointerException (no error message)

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 6s

    at makeError (C:\Users\Kimar\Projects\intentional-walk\node_modules\@react-native-community\cli-platform-android\node_modules\execa\index.js:174:9)
    at C:\Users\Kimar\Projects\intentional-walk\node_modules\@react-native-community\cli-platform-android\node_modules\execa\index.js:278:16      
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async runOnAllDevices (C:\Users\Kimar\Projects\intentional-walk\node_modules\@react-native-community\cli-platform-android\build\commands\runAndroid\runOnAllDevices.js:94:5)
    at async Command.handleAction (C:\Users\Kimar\Projects\intentional-walk\node_modules\@react-native-community\cli\build\index.js:186:9)        
info Run CLI with --verbose flag for more details.

**Comment by francisli** [comment]
@markkhoo I also see this message in the output:

```
> Configure project :react-native-reanimated
No AAR for react-native-reanimated found. Attempting to build from source.
```

And I found a corresponding issue in Github for it from a Google search:

https://github.com/software-mansion/react-native-reanimated/issues/2650

I'm going to try pinning the version of react-native-reanimated to an older version to see if that addresses the issue...

**Comment by markkhoo** [comment]
So I started pinning versions of dependencies that we're giving me errors until I ran into an error with my JDK

**Comment by francisli** [comment]
I'm seeing the Javac compilation errors now, I'm trying a fix outlined here:

https://github.com/facebook/react-native/issues/35210

TLDR; the React Native team broke all older builds when they released 0.70 and everyone needs to do a hotfix/patch.

**Comment by KaviNik** [comment]
info Installing the app...

> Configure project :react-native-reanimated
No AAR for react-native-reanimated found. Attempting to build from source.
Android gradle plugin: 7.4.2
Gradle: 8.0.1

> Task :react-native-reanimated:buildCMakeDebug[arm64-v8a] FAILED
